src/sgame/sg_bot_ai.cpp

```diff
@@ -1296,23 +1296,23 @@ AINodeStatus_t BotActionHealH( gentity_t *self, AIGenericNode_t *node )
 
 // triggers a reload.
 // Fails if there is no remaining clips.
-// TODO: return STATUS_RUNNING as long as reload is not
-//   finished (this would allow entering a more "dancing"
-//   stance for the bot, to be more careful, etc)
+// Keeps running as long as weapon is not ready.
 // TODO: make Cmd_Reload_f able to report status (that's
 //   true for all player commands actually, to allow bots
 //   to use them more easily)
 AINodeStatus_t BotActionReload( gentity_t *self, AIGenericNode_t *node )
 {
 	ASSERT( self && self->client );
+	playerState_t const& ps = self->client->ps;
 
-	if ( self->client->ps.clips == 0 )
+	if ( ps.clips == 0 )
 	{
 		return STATUS_FAILURE;
 	}
 
 	Cmd_Reload_f( self );
-	return STATUS_SUCCESS;
+
+	return ps.IsWeaponReady() ? STATUS_SUCCESS : STATUS_RUNNING;
```

Comment:
As discussed on IRC, I don't think it makes since for the reload action to have a "running" state, since any handling of this couldn't be applied to naturally occurring reloads when a clip runs out.

And dodging behavior would only be useful while fighting, but it would be stupid to reload in a fight with a non-empty clip.

Reply:
Example of use: you just killed an alien close to a door. If you reload instantly, you'll just continue to move on, opening the door (which are neat places to put defenses) while reloading.
You should more likely reload at safe distance of door and, when the weapon is ready, go open the door fully prepared.
Ofc I'm talking about bots being able to detect doors, which is currently not possible, since the navmesh generator just mostly ignore them, instead of creating offmesh links with the info as it should, but I can't address all problems at the same time, and I'd rather have things done once, or at least prepare for later improvements as much as possible.

Now, I agree that a better solution would likely be to check if the weapon (which one, actually? Humans can have up to 3 weapons, and aliens have only one, but some of it's modes might be ready or not at some time, effectively acting as if there were 3 weapons) is ready.
In the meantime, the code this PR does seems to work rather nicely. Seems, because since recently I'm starting to smell a bug in it, that I suspected in another part of my AI changes. Need more investigations, as this is pretty difficult (as always with AIs) to reproduce.

Reply:
It seems better to have a condition function for WeaponIsReady. Then you could detect any reload, not only a BT-requested one. If BT users wanted to reload without waiting, they could do that. And if you wanted this behavior of waiting until reload is finished, you could make a `sequence` with `reload` and then `condition !weaponIsReady { <something>} `

Reply:
> and then condition `!weaponIsReady { <something>}`

Ok, but doing that is quite fragile, because `playerState_t::IsWeaponReady()` do not apply to aliens. This is not too much important currently, since aliens won't reload anyway (none of them have clips), but I remember seeing cases in which aliens have only some of their "modes" not ready. IIRC this applies to `mantis` which can pounce between attacks and advanced dragoons with their barbs.
Of course, here, the correct solution would be to allow aliens to hold multiple weapons instead of considering that everything is the same weapon.
This remark also applies to the barbs being both infinite but with limited number of ammo.